### ISSUE TITLE
Initial Code Review

### ISSUE THREAD

**Issue by stuartromanek** [apostrophecms-legacy/apostrophe-link-preview-widgets#1]
Here it is

**Comment by boutell** [comment]
I love the configuration, flexibility and general syntax as a consumer after reading the README.

I would like to see a way to merge some stuff down, "body should be this or this or this, in that order," during configuration so that you don't have to write a lot of "or" statements in templates. Maybe even a standard config for some of that.

In fact maybe scrapers should have access to the complete nested data object so far, so that a scraper can be written just to act as a "merger."

`async-request` is not actively maintained (4 years since last commit). I would use `request-promise`. Since it returns a promise, you can still `await` it just the same. And, it turns HTTP errors into actual catch() errors like we all wish `request` did natively. Huzzah.

Or use `axios` if you really want the new hotness.

Change `scrapers` variable to `self.scrapers` so that it can be manipulated in an `improve` module or project level enhancement of this module.

Things get a little strange once we get into the load route. You are not using `async` functions and `await` statements consistently, for instance everywhere you have `.then()` you should be using `await` instead to just get the "return" value of that async function. Note this also works for functions like the cache "get" function which "return a promise" (because that's all async functions really do anyway, and "await" is just semantic sugar that gets rid of all that .then().catch() typing).

I would recommend writing an async method, `getResponse`, that gets the information from the cache or, if it can't find it there, gets it from the scrapers. Then `load` just `await`s that one method, and sends its return value. Instead of having the response sent in two places which makes it hard to follow the logic.

So...

* All ".then()" goes away
* Flatten things out, use "await" instead
* All the actual responding happens in one route function, which can be an async function - it's fine to write `async function(req, res) { ... }`

Also, right now you are not catch()ing any errors. But if you write your route function like this:

```
self.route('post', 'load', async function(req, res) {
   try {
     self.sendPreview(req, await self.getResponse());
   } catch (e) {
     self.apos.utils.error(e);
     self.sendError(req);
   }
});
```

then you don't have to worry about try/catch anywhere else, because it will all propagate up nicely to your `catch` statement, as long as you use async functions consistently.

**Comment by boutell** [comment]
(I made up sendError but any suitable response will do there.)

**Comment by boutell** [comment]
You might find that since they are only called from one place you can just inline the code for sendPreview and sendError at this point and it'll still read nice.

**Comment by stuartromanek** [comment]
refactor over here https://github.com/apostrophecms/apostrophe-link-preview-widgets/pull/2
